token/program-2022/src/extension/confidential_transfer/instruction.rs

```diff
@@ -551,39 +555,34 @@ pub fn configure_account(
     mint: &Pubkey,
     decryptable_zero_balance: AeCiphertext,
     maximum_pending_balance_credit_counter: u64,
-    context_state_account: Option<&Pubkey>,
     authority: &Pubkey,
     multisig_signers: &[&Pubkey],
-    proof_data: Option<&PubkeyValidityData>,
+    proof_data_location: ProofLocation<PubkeyValidityData>,
 ) -> Result<Vec<Instruction>, ProgramError> {
-    if let Some(proof_data) = proof_data {
-        Ok(vec![
-            inner_configure_account(
-                token_program_id,
-                token_account,
-                mint,
-                decryptable_zero_balance,
-                maximum_pending_balance_credit_counter,
-                None,
-                authority,
-                multisig_signers,
-                1,
-            )?,
-            verify_pubkey_validity(None, proof_data),
-        ])
-    } else {
-        Ok(vec![inner_configure_account(
-            token_program_id,
-            token_account,
-            mint,
-            decryptable_zero_balance,
-            maximum_pending_balance_credit_counter,
-            context_state_account,
-            authority,
-            multisig_signers,
-            0,
-        )?])
-    }
+    let mut instructions = vec![inner_configure_account(
+        token_program_id,
+        token_account,
+        mint,
+        decryptable_zero_balance,
+        maximum_pending_balance_credit_counter,
+        authority,
+        multisig_signers,
+        proof_data_location,
+    )?];
+
+    if let ProofLocation::Instruction(proof_instruction_offset, proof_data) = proof_data_location {
+        // This constructor appends the proof instruction right after the `ConfigureAccount`
+        // instruction. This means that the proof instruction offset must be always be 1. To
+        // use an arbitrary proof instruction offset, use the `inner_configure_account`
+        // constructor.
+        let proof_instruction_offset: i8 = proof_instruction_offset.into();
+        if proof_instruction_offset != 1 {
+            return Err(ProgramError::InvalidArgument);
```

Comment:
Can you add another error for this case since it's very specific? I agree with the behavior, but for someone calling this from the outside, it'll be much clearer to get a more special error that they can eventually look up to find out that they need to use `1`